deploy/chart/templates/plugin.yaml

```diff
@@ -1,17 +1,21 @@
-{{- if .Values.plugin.enabled -}}
+{{- range $key, $nodepool := .Values.nodepools -}}
+{{- if and $.Values.plugin.enabled (ne $nodepool nil) }}
+{{- $nodepool := mergeOverwrite $.Values $nodepool }}
```

Comment:
I think this is still modifying `$.Values`, which is not desired. Why not `merge $nodepool $.Values`?

Reply:
When I use merge, I get an unexpected result which is not same as official doc. So I choose to use mergeOverwrite.

Reply:
So if I use mergeOverwrite, It's still necessary to deepCopy `.Values`.

Reply:
OK, so there is some unexpected behavior here https://github.com/Masterminds/sprig/issues/255
Please proceed with deepCopy + mergeOverwrite

Reply:
done